### ISSUE TITLE
Latest master does not build for Android

### ISSUE THREAD

**Issue by leetal** [Microsoft/cpprestsdk#907]
Android builds does not work (with Clang++) on the current master due to missing definition of `abort_if_no_jvm()`.

https://github.com/Microsoft/cpprestsdk/blob/b9dd8eab814892f110853be1098df1b3f77b2ff0/Release/src/pplx/threadpool.cpp#L41

Problem is due to the fact that  `abort_if_no_jvm()` is declared in an anonymous namespace and has no prior declarations before `get_jvm_env()` uses it.

Solution is to just move the definition of  `abort_if_no_jvm()` above `get_jvm_env()`.

**Comment by BillyONeal** [comment]
Yeah, this one's my fault. That's what we get for having no continuous integration coverage for Android :(. Attempting to address that now....

**Comment by BillyONeal** [comment]
Fixed in https://github.com/Microsoft/cpprestsdk/commit/328646af0b96f770bf0e05a87d9185b8a31f70f0

**Comment by McMagnus** [comment]
Compiles, but doesn't link for Android.

commit 7467b79307e112ed130be54b28a587674365c66d broke it for me (using the embedded websocketpp). I need to add the crossplat namespace to use it.

**Comment by BillyONeal** [comment]
@McMagnus That suggests you are linking with an old build? That variable is not declared in a crossplat:: namespace: https://github.com/Microsoft/cpprestsdk/blob/7467b79307e112ed130be54b28a587674365c66d/Release/src/pplx/threadpool.cpp#L26

**Comment by McMagnus** [comment]
Yes it is. It is *declared* in the crossplat namespace in threadpool.h, and also *accessed* in that namespace from websockets/client/ws_client_wspp.cpp.

The file you pointed out shows that it isn't *implemented* in that namespace. All this means that it will compile ok but not link.
